lib/bundle.js

```diff
@@ -20,158 +20,253 @@ var Builder = require('systemjs-builder');
 var systemBuilder = new Builder();
 var fs = require('fs');
 var asp = require('rsvp').denodeify;
+var glob = require('glob');
 
 exports.depCache = function(moduleName) {
   return config.load()
-  .then(function() {
-    moduleName = moduleName || config.loader.main;
-  })
-  .then(function() {
-    ui.log('info', 'Injecting the traced dependency tree for `' + moduleName + '`...');
-
-    // trace the starting module
-    var cfg = config.loader.getConfig();
-    cfg.baseURL = path.relative(process.cwd(), config.pjson.baseURL);
-
-    return systemBuilder.trace(moduleName, cfg);
-  })
-  .then(function(output) {
-    var traceTree = output.tree;
-    moduleName = output.moduleName;
-    var depCache = config.loader.depCache = config.loader.depCache || {};
-    for (var d in traceTree) {
-      var deps = traceTree[d].deps.map(function(dep) {
-        return traceTree[d].depMap[dep];
-      });
+    .then(function() {
+      moduleName = moduleName || config.loader.main;
+    })
+    .then(function() {
+      ui.log('info', 'Injecting the traced dependency tree for `' + moduleName + '`...');
 
-      if (deps.length)
-        depCache[d] = deps;
-    }
-  })
-  .then(config.save)
-  .then(function() {
-    ui.log('ok', 'Depenency tree injected');
-  })
-  .catch(function(e) {
-    ui.log('err', e.stack || e);
-  });
+      // trace the starting module
+      var cfg = config.loader.getConfig();
+      cfg.baseURL = path.relative(process.cwd(), config.pjson.baseURL);
+
+      return systemBuilder.trace(moduleName, cfg);
+    })
+    .then(function(output) {
+      var traceTree = output.tree;
+      moduleName = output.moduleName;
+      var depCache = config.loader.depCache = config.loader.depCache || {};
+      for (var d in traceTree) {
+        var deps = traceTree[d].deps.map(function(dep) {
+          return traceTree[d].depMap[dep];
+        });
+
+        if (deps.length)
+          depCache[d] = deps;
+      }
+    })
+    .then(config.save)
+    .then(function() {
+      ui.log('ok', 'Depenency tree injected');
+    })
+    .catch(function(e) {
+      ui.log('err', e.stack || e);
+    });
 };
 
 function extractOperations(args) {
+  var i;
   var operations = [];
 
-  for (var i = 1; i < args.length - 1; i = i + 2) {
-    var operator = args[i];
-    var moduleName = args[i + 1];
-
-    operations.push({
-      operator: operator,
-      moduleName: moduleName
+  var seq = expandGlob(args[0])
+    .then(function(moduleNames) {
+      pushOps(moduleNames, '+');
```

Comment:
I think this is what I have done. I may not understand your comment though. Could you please take a look at it again. Your example should work too.